plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -121,6 +131,21 @@ public ConnectorSplitSource getSplits(
         return new ClassLoaderSafeConnectorSplitSource(splitSource, DeltaLakeSplitManager.class.getClassLoader());
     }
 
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, SchemaFunctionName name, ConnectorTableFunctionHandle function)
+    {
+        if (name.equals(TABLE_CHANGES_NAME)) {
+            TableChangesTableFunctionHandle tableFunctionHandle = (TableChangesTableFunctionHandle) function;
+            DeltaLakeMetastore metastore = getMetastore(session, transaction);
+            SchemaTableName schemaTableName = new SchemaTableName(tableFunctionHandle.getSchemaName(), tableFunctionHandle.getTableName());
+            if (tableFunctionHandle.getSinceVersion() > metastore.getSnapshot(schemaTableName, session).getVersion()) {
```

Comment:
Can't we add the table version / make this check in `io.trino.plugin.deltalake.functions.tablechanges.TableChangesFunction#analyze` to avoid making this call at this level?

Reply:
and return what from `io.trino.plugin.deltalake.functions.tablechanges.TableChangesFunction#analyze` ?
It is not an exceptional situation, it is a simple situation there is no data, that's all.

SplitManager looks like a good place to return emptySplitSource, it is returned  also returned in the same class here https://github.com/trinodb/trino/blob/e7d560ff347e242215706171b17590befcaa5a3d/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java#L109